Tests/MongoSwiftSyncTests/TransactionsTests.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import MongoSwift
+import Nimble
+import TestsCommon
+
+/// Struct representing a single test within a spec test JSON file.
+private struct TransactionsTest: SpecTest {
+    let description: String
+
+    let operations: [TestOperationDescription]
+
+    let outcome: TestOutcome?
+
+    let skipReason: String?
+
+    let useMultipleMongoses: Bool?
+
+    let clientOptions: TestClientOptions?
+
+    let failPoint: FailPoint?
+
+    let sessionOptions: [String: ClientSessionOptions]?
+
+    let expectations: [TestCommandStartedEvent]?
+
+    static let sessionNames: [String] = ["session0", "session1"]
+}
+
+/// Struct representing a single transactions spec test JSON file.
+private struct TransactionsTestFile: Decodable, SpecTestFile {
+    private enum CodingKeys: String, CodingKey {
+        case name, runOn, databaseName = "database_name", collectionName = "collection_name", data, tests
+    }
+
+    let name: String
+
+    let runOn: [TestRequirement]?
+
+    let databaseName: String
+
+    let collectionName: String?
+
+    let data: TestData
+
+    let tests: [TransactionsTest]
+}
+
+final class TransactionsTests: MongoSwiftTestCase, FailPointConfigured {
+    var activeFailPoint: FailPoint?
+
+    override func tearDown() {
+        self.disableActiveFailPoint()
+    }
+
+    override func setUp() {
+        self.continueAfterFailure = false
+    }
+
+    func testTransactions() throws {
+        let skippedTestKeywords = [
+            "count", // skipped in RetryableReadsTests.swift
+            "mongos-pin-auto", // useMultipleMongoses, targetedFailPoint not implemented
```

Comment:
do we need a separate ticket tracking these?

Reply:
I think we need a separate ticket tracking the `useMultipleMongoses` part.

Reply:
ok let's create a ticket then - and I think I answered that in my comment below
> so the reason we skip this in retryable reads is that the tests are written using the old count API, which was last minute deprecated right before the 4.0 release. we don't have the old API at all but some older drivers that haven't tagged a major version since then still do.

Reply:
Got it. I've actually implemented support for `useMultipleMongoses` and `targetedFailPoint`, which I've included in the most recent commit. Evergreen is still failing with some weird errors so I will comment out the tests for now. I'll create a ticket to remind us to finish this.

Reply:
Created [SWIFT-774](https://jira.mongodb.org/browse/SWIFT-774).